students/paul_jurek/lesson2/grid_printer.py

```diff
@@ -0,0 +1,79 @@
+def grid_printer():
+	"""print grid as part of python 201 lesson 2"""
+	print('+','-'*4,'+','-'*4,'+',sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('+','-'*4,'+','-'*4,'+',sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('|', ' '*4, '|', ' '*4, '|', sep='')
+	print('+','-'*4,'+','-'*4,'+',sep='')
+	
+	
+def print_grid(grid_size:int):
+	"""defines dyamic print_grid
+	args:
+		n: number ofr grid elemnts verticle and horizonatal
+			if odd, rounds down.  
+	returns:
+		na
+	"""
+	mark_counter = grid_size//2
+	def make_border(mark_counter):
+		"""helper function to make top, mid and bottom"""
+		print('+','-'*mark_counter,'+','-'*mark_counter,'+',sep='')
+	
+	def make_middle(mark_counter):
+		count = 0
+		while (count<mark_counter):
+			print('|', ' '*mark_counter, '|', ' '*mark_counter, '|', sep='')
+			count += 1
```

Comment:
no reason to define these as inner functions, especially if you're passing in `mark_counter` anyway.